orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -174,7 +174,7 @@ class BakeStage implements StageDefinitionBuilder {
 
       if (failOnImageNameMismatchEnabled()) {
         // find distinct image names in bake stages that are actually related to the stage passed into the task
-        List<String> distinctImageNames = relatedBakeStages
+        List<Object> distinctImageNames = relatedBakeStages
           .findAll { childStage -> childStage.parentStageId == stage.id && childStage.context.imageName }
           .stream()
           .map { childStage -> childStage.context.imageName }
```

Comment:
Aaah, ok.  Looks like we're running into the snag where [StageExecution.getContext()](https://github.com/spinnaker/orca/blob/2e8ebf811073b3c2100712e16a2cf68fb675ec00/orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/models/StageExecution.java#L85) returns a Map<String, Object>, so unless we cast `childStage.context.imageName` to a String, Object is the best we can do.